uportal-war/src/main/java/org/jasig/portal/groups/smartldap/SmartLdapGroupStore.java

```diff
@@ -604,6 +642,59 @@ private GroupsTree buildGroupsTree() {
 
         }
 
+        // Process each group in order to create the groups that will represent the Grouper's folders
+        // and maintain the consistency regarding the membership.
+        for (String groupKey : new_groups.keySet()) {
+
+            boolean stop = false;
+            boolean isAFolder = false;
+            String currentPath = groupKey;
+            while (!stop) {
+                final String containingFolder = getContainingFolder(currentPath);
+
+                if (isAFolder) {
+                    // The group that represents the grouper folder is created.
+                    final IEntityGroup folderGroup = new EntityTestingGroupImpl(currentPath, IPerson.class);
+                    folderGroup.setCreatorID("System");
+                    folderGroup.setName(currentPath);
+                    folderGroup.setDescription("Grouper Folder");
+                    new_folders.put(currentPath, folderGroup);
+                } else {
+                    // The first entry is the group itself.
+                    isAFolder = true;
+                }
+
+                if ("".equals(containingFolder)) {
```

Comment:
I know but doing this check in this way avoid NPE in main case. On an other side should I check if is null !
If not since java 6 we can use .isEmpty() on String, of from apache.common.lang there is a method to check nullOrEmpty... I will check what is the best.